### PR TITLE
[usage] Configure ingress for Stripe webhook

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description

Configures ingress between `proxy` and the public API server to allow external access to the Stripe webhook (added in https://github.com/gitpod-io/gitpod/pull/11806).

* Add a new `backend` section to the `payment-ingress` section in the proxy `Caddyfile` and 
* Update a `networkpolicy` to allow `proxy` to reach the `public-api-server`.

## Related Issue(s)

Part of https://github.com/gitpod-io/gitpod/issues/10937

## How to test

* Hit the webhook handler using its external URL:
```
brew install httpie
http POST https://payment.af-ingressf994192aa3.preview.gitpod-dev.com/stripe/invoices/webhook type=foo
```
Expected output:

```http
HTTP/1.1 200 OK
Content-Length: 15
Content-Type: text/plain; charset=utf-8
Date: Thu, 04 Aug 2022 08:05:21 GMT
X-Gitpod-Region: local.

event type: foo
```

## Release Notes
```release-note
NONE
```

## Werft options:
- [x] /werft with-preview

### CODE REVIEW

File: install/installer/cmd/testdata/render/minimal/output.golden [hunk 1]
```diff
@@ -1424,8 +1424,7 @@ data:
           "gcloud": {
             "credentialsFile": "",
             "region": "",
-            "projectId": "",
-            "maximumBackupCount": 0
+            "projectId": ""
```
Comment (easyCZ) [inline install/installer/cmd/testdata/render/minimal/output.golden:1427]: Is this related to your change?

**Review by easyCZ — APPROVED** [review]
LGTM

/hold
for Q about generated file

File: install/installer/cmd/testdata/render/minimal/output.golden (hunk 1 above)
Comment (andrew-farries) [reply install/installer/cmd/testdata/render/minimal/output.golden:1427]: Nope. The file was regenerated with `make generateRenderTests` following the [README](https://github.com/gitpod-io/gitpod/tree/main/install/installer/cmd/testdata/render). 

I don't know what accounts for the changes in the file besides the change to the proxy config. Maybe the file was not updated by a previous PR?

Comment (easyCZ) [reply install/installer/cmd/testdata/render/minimal/output.golden:1427]: I guess Self-Hosted is on the review and can clarify. Perhaps we're missing a CI check that would fail if the `output.golden` doesn't match anymore.

**Review by MrSimonEmms — APPROVED** [review]
The changes look good and happy to approve. We've just had a change merged which updated the render golden files so probably worth rebasing (again) as it's likely to fail when merged to `main`.

Remove the hold at will

/hold

**Comment by andrew-farries** [comment]
Rebased and tests pass so

/unhold
